evalml/automl/utils.py

```diff
@@ -117,3 +112,29 @@ def check_all_pipeline_names_unique(pipelines):
 AutoMLConfig = namedtuple("AutoMLConfig", ["ensembling_indices", "data_splitter", "problem_type",
                                            "objective", "additional_objectives", "optimize_thresholds",
                                            "error_callback", "random_seed"])
+
+
+def get_best_sampler_for_data(X, y, sampler_type, sampler_balanced_ratio):
+    """Returns the name of the sampler component to use for AutoMLSearch.
+
+    Arguments:
+        X (ww.DataTable): The input feature data
+        y (ww.DataColumn): The input target data
+        sampler_type (str): The sampler_type argument passed to AutoMLSearch
+        sampler_balanced_ratio (float): The ratio of min:majority targets that we would consider balanced,
+            or should balance the classes to.
+
+    Returns:
+        str: The string name of the sampling component to use
+    """
+    if sampler_type != 'auto':
+        return sampler_type
```

Comment:
To keep things simple, let's delete this, delete the `sampler_type` arg to this method and instead add this to automl search init where this method gets called:
```python
if self.sampler_method == 'auto':
    self._sampler_name = get_best_sampler_for_data(self.X_train, self.y_train, self.sampler_balanced_ratio)
```
This way, `get_best_sampler_for_data` stays simple, easier to test, and we can have separate automl search mocked testing to cover the sampler "auto" mode.

Update: ah yes I see your excellent test `test_automl_search_sampler_method` provides coverage for this, so that's great!